### PR TITLE
Plots get labels from pint arrays

### PR DESCRIPTION
**Pull request by TomNicholas**
Stops you needing to call `.pint.dequantify()` before plotting.

Builds on top of #5568, so that should be merged first.

- [x] Closes (1) from https://github.com/pydata/xarray/issues/3245#issue-484240082
- [x] Tests added
- [x] Tests passing
- [x] Passes `pre-commit run --all-files`
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Comment by TomNicholas** [comment]
I thought this approach was neat but it seems to have broken some of the tests in `test_units`...

**Comment by TomNicholas** [comment]
This should be reimplemented following https://github.com/pydata/xarray/issues/3245#issuecomment-662547659

**Comment by TomNicholas** [comment]
Now builds on top of #5568, but fails because that PR is not complete (does not change `.values` -> `.to_numpy()` everywhere needed yet).

File: xarray/tests/test_units.py
Comment (TomNicholas) [reply xarray/tests/test_units.py:5585]: The point was that I want this test to eventually test labels on both the x and y axes, but at the moment pint is only involved with the y axis, right?

File: xarray/core/pycompat.py
Comment (TomNicholas) [reply xarray/core/pycompat.py:54]: Do you think this explains the failures in #5568 too?

File: xarray/tests/test_units.py
Comment (keewis) [reply xarray/tests/test_units.py:5585]: that's right. I meant to implement this by converting the dimension coordinate to a non-dimension coordinate:
```python
x_coord = xr.DataArray(
    np.linspace(1, 3, 3) * unit_registry.m, dims="x"
)
da = xr.DataArray(data=arr, dims="x", coords={"x_coord": x_coord}, name="pressure")
da.plot.line(x="x_coord")
```

Comment (TomNicholas) [reply xarray/tests/test_units.py:5585]: If I try that I still get a `UnitStrippedWarning` in `.to_index_variable()`

File: xarray/core/pycompat.py
Comment (TomNicholas) [reply xarray/core/pycompat.py:54]: Looking at this properly I see what you mean now. So is this PR just stuck then? Why can we not just check the name rather than the explicit type within xarray (for now at least)?

Comment (keewis) [reply xarray/core/pycompat.py:54]: you could import `pint` in the functions that use `pint_array_type` for now, and I'll try to get `pint` to not import `xarray` (because ideally it should not care about what wraps it). The best long term fix would be to figure out dask/dask#6635, though.

Comment (jthielen) [reply xarray/core/pycompat.py:54]: @keewis It would be good to hear your thoughts on that over on the Pint issue tracker. Right now, to maintain the type casting hierarchy, libraries with duck arrays have to either define a list of types they can wrap (and prohibit all others) or a list which are meant to wrap it (and allow wrapping any other). Pint went with the latter since it was the safest approach as a type high on the graph (https://pint.readthedocs.io/en/stable/numpy.html#Technical-Commentary), but I suppose that if that was problematic, it could be flipped around to a list of allowed types to wrap (with a breaking change), as Dask ended up doing (https://github.com/dask/dask/pull/6393).

But, definitely, this ad hoc approach to the type casting hierarchy will continue to be problematic until https://github.com/dask/dask/issues/6635 is resolved definitely.

Comment (TomNicholas) [reply xarray/core/pycompat.py:54]: It would be really nice to fix this properly (and I will bring it up in the xarray dev call now), but for this PR then your suggestion seems to work nicely @keewis , thanks.

**Comment by TomNicholas** [comment]
I think this is ready to merge after #5568 is merged?

File: xarray/tests/test_units.py
Comment (dcherian) [reply xarray/tests/test_units.py:5585]: That seems like a bug. open an issue for it so we don't forget?

**Review by dcherian — APPROVED** [review]
looks great!

**Pull request merged**
